### ISSUE TITLE
Bug: `id-length` doesn't throw error for characters consisting of two code units

### ISSUE THREAD

**Issue by sosukesuzuki** [eslint/eslint#16316]
### Environment

Node version: v16.14.2
npm version: v8.5.0
Local ESLint version: v8.23.1 (Currently used)
Global ESLint version: Not found
Operating System: darwin 21.6.0

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```
/*eslint id-length: "error"*/
```
</details>

[ESLint Playground](https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgaWQtbGVuZ3RoOiBcImVycm9yXCIqL1xuXG5cbi8vIGludmFsaWRcbmNvbnN0IGEgPSAyO1xuLy8gaW52YWxpZFxuY29uc3Qg5Y+xID0gMjtcbi8vIHZhbGlkP1xuY29uc3Qg8KCunyA9IDI7XG4iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOiJsYXRlc3QiLCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19)

```js
// invalid
const a = 2;
// invalid
const 叱 = 2;
// valid?
const 𠮟 = 2;

```

### What did you expect to happen?

Error is reported from ESLint.

### What actually happened?

Ignored.

### Participation

- [X] I am willing to submit a pull request for this issue.

### Additional comments

The length of an identifier consisting of two code units cannot be correctly treated.

**Comment by mdjermanovic** [comment]
Thanks for the issue!

This looks like a bug to me, but I'd like more opinions from other team members before accepting this issue.

I think we should either count code points instead of code units, or use [`grapheme-splitter`](https://github.com/eslint/eslint/blob/8cc0bbe440dc5e6af6ef02f00d0514a40ca07c24/package.json#L80) here.

**Comment by snitin315** [comment]
Yes, Definitely a bug! 

+1 for using [grapheme-splitter](https://github.com/eslint/eslint/blob/8cc0bbe440dc5e6af6ef02f00d0514a40ca07c24/package.json#L80).
